### ISSUE TITLE
Documentation unclear about the meaning of high availability

### ISSUE THREAD

**Issue by evitalis** [hashicorp/vault#7155]
**Describe the bug**

Documentation linked below talks about high availability and that only certain backends support this by means of a lock file. But it does not indicate what the purpose of said lockfile is, or what active/standby means in terms of Vault.

From an ops perspective multiple Vault nodes behind a load balancer pointing to a single database can be considered highly available but this doesn't seem to match the documentation.

https://www.vaultproject.io/docs/concepts/ha.html

Is it possible this is simply unclear to try and encourage the usage of Consul?

**To Reproduce**
1. Reading the documentation

**Expected behavior**
Explanation of what active/standby high availability means in the world of Vault, especially when using non-Consul backends. The documentation is very vague and does not discuss why this is useful or define the terms within the document.

**Environment:**
* Vault Server Version (retrieve with `vault status`): all
* Vault CLI Version (retrieve with `vault version`): all
* Server Operating System/Architecture: any

Vault server configuration file(s):

```hcl
n/a
```

**Additional context**
n/a at this time

**Comment by jefferai** [comment]
That documentation is vague because the specifics are dependent on the storage backend. If it satisfies an interface, it can be used for HA, but the mechanism is internal to that backend. That's why further details are left to the docs for a given backend.

**Comment by evitalis** [comment]
@jefferai This should not have been closed, an entire part of the question was ignored. What is Vault HA as defined in the documentation vs the example I provided which may or may not do the same thing.

Much of the documentation is confusingly written or very vague for no apparent reason, and can put people off from using the product.

**Comment by jefferai** [comment]
I guess I'm not sure what is lacking. The documentation goes into great detail as to how it works and how secondaries behave.

**Comment by fernando-villalba** [comment]
I agree with @evitalis I am trying to get more clarity on this right now and I am just as confused as he/she is/was. 

For example right now I am looking at one use case where consul is used as a storage backend to "manage the lock" and S3 to manage persistent data:

https://github.com/hashicorp/terraform-aws-vault/tree/master/examples/vault-s3-backend

I am guessing what that means is that in s3 you have no way to prevent two processes to write to the same place but in consul you do? It is not very well explained. 

Also where does that leave consul? Does that mean you can and should still do snapshots of consul to manage your backups or does it mean that also using s3 as a backend you cannot do this and you need to backup by copying files from the bucket itself?

In a nutshell, is data stored at the same time in both bucket and consul? 

Of course I could always start spinning clusters and experimenting until I find the answers myself, but the better explained it is on the documentation, the more time it saves everyone.

**Comment by kalafut** [comment]
>In a nutshell, is data stored at the same time in both bucket and consul?

From the docs:
>Vault also supports split data/HA mode, whereby the lock value and the rest of the data live separately.

The lock value must be written to an HA-capable store in order to allow the HA leader election process to reliably work (i.e. we don't want more than one leader, or no leader). As Jeff said, the way this occurs is different per storage backend based on what operations they support to achieve the lock interface. The "rest" of the data really is all of the other Vault data. So in your example all Vault data, except for the lock, would only in S3. A backup would be taken from S3. The lock value needn't be backed up.

**Comment by fernando-villalba** [comment]
Thank you @kalafut That does clarify it further. Have you worked with this mode yourself? I am pondering which one would be easier to upgrade in an automated way without downtime. I have managed to do this quite nicely for the vault cluster itself with the method described here: 

https://groups.google.com/forum/#!msg/terraform-tool/7Gdhv1OAc80/iNQ93riiLwAJ

But doing the above when consul is the main backend would not do without a significant downtime and some manual work. Do you think  it would be feasible to do A/B upgrading of this kind when consul is only used for lock storage or would there be risk of data loss?

**Comment by kalafut** [comment]
Hi @fernando-villalba. I'd suggest posting your setup and question on https://discuss.hashicorp.com/c/vault. That will get a lot more eyes than this GH issue.
